agent.go

```diff
@@ -53,33 +53,36 @@ type process struct {
 type container struct {
 	sync.RWMutex
 
-	id          string
-	initProcess *process
-	container   libcontainer.Container
-	config      configs.Config
-	processes   map[string]*process
-	mounts      []string
+	id              string
+	initProcess     *process
+	container       libcontainer.Container
+	config          configs.Config
+	processes       map[string]*process
+	mounts          []string
+	useSandboxPidNs bool
 }
 
 type sandbox struct {
 	sync.RWMutex
 
-	id              string
-	running         bool
-	noPivotRoot     bool
-	enableGrpcTrace bool
-	containers      map[string]*container
-	channel         channel
-	network         network
-	wg              sync.WaitGroup
-	sharedPidNs     namespace
-	mounts          []string
-	subreaper       reaper
-	server          *grpc.Server
-	pciDeviceMap    map[string]string
-	deviceWatchers  map[string](chan string)
-	sharedUTSNs     namespace
-	sharedIPCNs     namespace
+	id               string
+	running          bool
+	noPivotRoot      bool
+	enableGrpcTrace  bool
+	containers       map[string]*container
+	channel          channel
+	network          network
+	wg               sync.WaitGroup
+	sharedPidNs      namespace
+	mounts           []string
+	subreaper        reaper
+	server           *grpc.Server
+	pciDeviceMap     map[string]string
+	deviceWatchers   map[string](chan string)
+	sharedUTSNs      namespace
+	sharedIPCNs      namespace
+	infraContainerId string
```

Comment:
I don't think you need the `infraContainerId` field. You can determine if the container is the very first container in a sandbox and use it to set the sandbox level pidns path if `sandboxPidNs` is false.

Reply:
Yes, I just added this field to make this explicit. I can take this out.

Reply:
Thanks!

Reply:
Fixed.